### DISCUSSION TITLE
ep760

### DISCUSSION THREAD

**Discussion by JADMIM** [Patrick762/hassio-bluetti-bt#34] [General]
hello,

I have seen the integration of the bluetti with home assistant, how I could do to integrate the EP760 model.
Thank you so much

**Comment by Patrick762** [comment]
Seems like someone already did the discovery here: https://github.com/warhammerkid/bluetti_mqtt/issues/91 (bluetti_mqtt seems to be no longer maintained)
I found some values in the App, but it also seems like they will change to a V2 of the protocol soon. This could explain some other issues with the integration. I'll have to look into this a bit more.

**Reply by JADMIM** [reply]
Good afternoon,

I'm a little new to this, how should I integrate these codes that you gave me before the integration into my home assistant? Thank you so much

**Reply by Patrick762** [reply]
Hello @JADMIM 
I found some more info on the bluetti protocol. They use 2 different versions, some devices with v1 and some v2. They most likely wont switch every device to v2.

Because they only use 2 protocols and the EP700 is relatively new, it most likely uses v2 (same as EP600). I'll publish a beta version of the integration soon which includes a few sensors.

**Reply by Patrick762** [reply]
@JADMIM  https://github.com/Patrick762/hassio-bluetti-bt/discussions/38

**Reply by Patrick762** [reply]
Sorry, I added the wrong Powerstation. I'll rename it today and pubish a new beta.

**Reply by JADMIM** [reply]
Good afternoon,

If you can get my ep760 to read more data with the integration, tell me, I can try it and I'll tell you if it works and if you want me to do tests or if you need any information about my ep760, tell me and I can send you what I want. do missing.

Thank you

**Reply by mikwhe** [reply]
I finally managed to get my EP760 connected to Bluetti MQ/HA by copying the EP600.py file. 
I can confirm this is protocol 2 device.
The only registers I seem to be missing are for PV3/PV4(AC) input which doesn't matter to me as don't have anything connected.  
HTH

**Reply by joe6602** [reply]
I can confirm that the EP800 is a version 2 protocol device.  The other day, I did as mikwhe and copied the EP600 code into the ep800.py and now I have all the data.

**Reply by JADMIM** [reply]
hi,

I copied the code of the ep600 and I get an error, I don't know if I did it right, as you did. I have an ep760
thank you

**Reply by mikwhe** [reply]
You'll need to edit the new ep760.py file created and replace the text (near top) which will still refer to EP600 with EP760 value, below is example:

class EP760(ProtocolV2Device):
    def __init__(self, address: str, sn: str):
   super().__init__(address, "EP760", sn)
.....
Let me know if your still stuck and I'll upload full src for you to try

**Comment by JADMIM** [comment]
Good afternoon,

If you can get my ep760 to read more data with the integration, tell me, I can try it and I'll tell you if it works and if you want me to do tests or if you need any information about my ep760, tell me and I can send you what I want. do missing.

Thank you

**Comment by ciprianchiran** [comment]
Hello - Trying very hard to connect to my EP760 but always says that it has failed to setup "no devices or entities". I am using version 0.0.23 is that the best one for me to use? Thank you

**Comment by JADMIM** [comment]
Good morning,

Of all the latest versions, only the % battery charge data and the total amount of energy that my solar installation has produced have always appeared.
Kind regards

**Comment by mikwhe** [comment]
I've just managed to get more DATA (AC consumption, total PV input) from my EP760 by copying the existing ep600.py to ep760.py and then changing the device name in src.
Not sure this helps others but does confirm these new devices do share same v2 protocol.  
BTW, please keep up all your awesome work!!!

**Comment by joe6602** [comment]
Hey Patrick762, can you copy the ep600.py to the ep800.py as mikwhe describes above?

**Comment by Apfuntimes** [comment]
I have done that for my EP760 as well.

Now I have the majority of fields notable exceptions:
 - missing the 3rd DC/PV Solar Inputs  
 - no detailed battery information - just the SoC %
 - no inverter status - or operating mode for the overall EP760 (ie. Grid Connected/Off Grid etc)
 
Notable fields not required:
 - My EP760 is a single phase unit, thus no need for Phases 2 & 3 on the AC side at all - eliminating voltages, power and current.

Cheers,
AP

**Comment by joe6602** [comment]
For single phase, you need Phase 1 and Phase 2 information.
    On Thursday, October 3, 2024 at 06:56:47 AM PDT, Apfuntimes ***@***.***> wrote:  
 
 


I have done that for my EP760 as well.

Now I have the majority of fields notable exceptions:
   
   - missing the 3rd DC/PV Solar Inputs
   - no detailed battery information - just the SoC %
   - no inverter status - or operating mode for the overall EP760 (ie. Grid Connected/Off Grid etc)

Notable fields not required:
   
   - My EP760 is a single phase unit, thus no need for Phases 2 & 3 on the AC side at all - eliminating voltages, power and current.

Cheers,
AP

**Reply by Apfuntimes** [reply]
Ahh yes if you are in the US, here in. Aus, no split phase and only 240v. Similar to Europe.

so only 1 phase for 240v Au.

I should have  clarified that.

**Comment by joe6602** [comment]
Oh there I go, thinking like an American.  Sorry about that!
    On Thursday, October 3, 2024 at 12:56:22 PM PDT, Apfuntimes ***@***.***> wrote:  
 
 


Ahh yes if you are in the US, here in. Aus, no split phase and only 240v. Similar to Europe.

so only 1 phase for 240v Au.

I should have clarified that.

**Reply by Apfuntimes** [reply]
It was working, and now l suspect the latest patch on the ep760 or the app turned on some form of authentication…which hopefully we can replicate easily enough… I am already missing what I had for a fleeting few days…

**Reply by Apfuntimes** [reply]
So I have removed the BT password and then if I reset the IOT controller, the integration works for a bit...

Then the BT connection is dropped by the IOT controller I think and BT is no longer available from either HA or the Bluetti app until I reset the IOT controller... strange. 

I suspect there is some additional check by the IOT controller... or it thinks the BT request is malicious for some reason... I am connecting BT through an ESP32 BT proxy from HA.

**Reply by Apfuntimes** [reply]
For those wondering configuration options which work and don’t seem to trip the Bluetooth IOT are:
Polling interval: 30s
Polling timeout: 45s
Max retries: 3
Persistent connection: disabled
Use controls: disabled 

so it’s now up and running and pretty stable! 🤞
